### PR TITLE
DSG-1670: Drive form buttons with workflow actions

### PR DESCRIPTION
**Pull request by kevincheng0**
# Basics

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Styling and layout changes
- [ ] Code style changes (formatting, local variables)
- [ ] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [ ] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?
Renders the 'Submit' button on the public form using the actions from the backend.

## Screenshot of the affected changes
Everything looks the same as before, just now configurable from backend
image
image
image

## Checklist

- [ ] I have updated the documentation accordingly
- [ ] I have confirmed all acceptance criteria defined by the story have been met
- [ ] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

### CODE REVIEW

File: libs/shared/feature/form-nuv/src/lib/components/steps/formly/formly-steps.component.spec.ts [hunk 1]
```diff
@@ -107,22 +114,33 @@ const STEPS: IStep[] = [
...
-    components: [
-      ...mockPartialFields,
-      {
-        props: {
-          label: 'Review & Confirm',
-          stepKey: CONFIRMATION_STEP_KEY,
-        },
-      },
-    ],
+    components: [...mockPartialFields],
```
Comment (kevincheng0) [inline libs/shared/feature/form-nuv/src/lib/components/steps/formly/formly-steps.component.spec.ts:117]: Added an extra step that was not defined in `STEPS` messed up some unit tests

File: libs/shared/feature/form-nuv/src/lib/components/steps/formly/formly-steps.component.spec.ts [hunk 2]
```diff
@@ -250,17 +297,7 @@ describe('FormlyStepsComponent', () => {
...
-      const fieldConfig: any = mockPartialFields[1];
-      if (fieldConfig.components !== undefined) {
-        for (const group of fieldConfig.components) {
-          if (group.props !== undefined) {
-            group.props.required = false;
-          }
-        }
-      }
-    }
+    confirmationTestSetup();
```
Comment (kevincheng0) [inline libs/shared/feature/form-nuv/src/lib/components/steps/formly/formly-steps.component.spec.ts:300]: Is there a context manager similar to Python's `with` I can use here?

File: libs/shared/feature/form-nuv/src/lib/components/steps/formly/formly-steps.component.spec.ts [hunk 3]
```diff
@@ -551,4 +595,115 @@ describe('FormlyStepsComponent', () => {
...
+      const transactionModelWithAction = new TransactionModel({ ...transactionModelMock, activeTasks: [mockSubmitActiveTask] });
+      transaction.next(transactionModelWithAction);
+
+      component.loadFooterActions$.subscribe(() => {
+        expect(component.activeTaskActions).toEqual([mockSubmitAction]);
+      });
+    });
+  });
+
+  describe('footer action buttons', () => {
```
Comment (davidyoung77) [inline libs/shared/feature/form-nuv/src/lib/components/steps/formly/formly-steps.component.spec.ts:617]: These are good tests, thanks

**Review by davidyoung77 — COMMENTED** [review]
This is good code, I would like to look at the functionality with you before merging

File: libs/shared/feature/form-nuv/src/lib/components/steps/formly/formly-steps.component.ts [hunk 4]
```diff
@@ -62,6 +104,55 @@ export class FormlyStepsComponent extends FormlyBaseComponent implements OnInit
   public updatingTransaction = false;
   public formErrors$: Observable<IFormError[]> = this._formTransactionService.formErrors$;
 
+  public actions: FooterAction[] = [];
+  public loadFooterActions$ = this._formTransactionService.transaction$.pipe(
+    // We currently assume that there can be a maximum of one active task
+    tap(transaction => {
+      if (transaction.activeTasks.length && transaction.activeTasks[0].actions.length) {
+        // Only one action so far
+        this.activeTaskActions = transaction.activeTasks[0].actions;
```
[the remark this answers is not in the document]
Comment (kevincheng0) [reply libs/shared/feature/form-nuv/src/lib/components/steps/formly/formly-steps.component.ts:113]: `this.activeTaskActions` only contains the `actions` in the `activeTasks` of a transaction. The back button and save & continue button are stored in `baseFooterActions` and are added to `actions` upon startup/changing steps. Here I'm just storing the "submit" action after it's received

**Review by davidyoung77 — APPROVED** [review]
Good work approved

**Pull request merged**